test/jdk/jdk/incubator/vector/templates/Unit-header.template

```diff
@@ -742,8 +742,27 @@ public class $vectorteststype$ extends AbstractVectorTest {
 
 
 #if[FP]
-#if[!Halffloat]
     static boolean isWithin1Ulp($type$ actual, $type$ expected) {
+#if[Halffloat]
+        float act = Float.float16ToFloat(actual);
+        float exp = Float.float16ToFloat(expected);
+        if (Float.isNaN(exp) && !Float.isNaN(act)) {
+            return false;
+        } else if (!Float.isNaN(exp) && Float.isNaN(act)) {
+             return false;
+        }
+
+        float low = Math.nextDown(exp);
+        float high = Math.nextUp(exp);
+
```

Comment:
Here upcasting the result to float will check against the ULP of Float not Halffloat. But given that ULP of FP32 is always greater than ULP of FP16 intended functionality should still be honored.  

Another observation we should be using an expression like   Math.abs((expected + Math.ulp(expected)) - actual)) <= Math.ulp(expected), but this is not related to current patch, since patch just enable existing JTREG tests for FP16.

Reply:
Ok